server/capabilities/aks_version_endpoint.go

```diff
@@ -0,0 +1,164 @@
+package capabilities
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2017-09-30/containerservice"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/mcuadros/go-version"
+	"net/http"
+	"sort"
+)
+
+func NewAKSVersionsHandler() *AKSVersionHandler {
+	return &AKSVersionHandler{}
+}
+
+type AKSVersionHandler struct {
+}
+
+type regionCapabilitiesRequestBody struct {
+	// credentials
+	ClientID       string `json:"clientId"`
+	ClientSecret   string `json:"clientSecret"`
+	SubscriptionID string `json:"subscriptionId"`
+	TenantID       string `json:"tenantId"`
+
+	Region string `json:"region"`
+}
+
+func (g *AKSVersionHandler) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
+		writer.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+
+	writer.Header().Set("Content-Type", "application/json")
+
+	var body regionCapabilitiesRequestBody
+	err := extractRequestBody(writer, req, &body)
```

Comment:
Nit: this can be simplified as
```
if err := extractRequestBody(writer, req, &body); err != nil {
   handle(err)
}
``` 
This applies to other places